tests_e2e/playwright/navigator.py

```diff
@@ -0,0 +1,448 @@
+
+        """
+        if kernel is None:
+            # close dialog (deal with the two formats of this dialog)
+            try:
+                self.page.get_by_text("Cancel", exact=True).click()
+            except Exception:
+                self.page.locator("div").filter(has_text="No KernelSelect").get_by_role(
+                    "button", name="No Kernel"
+                ).wait_for(timeout=300, state="attached")
+        else:
+            # set the environment
+            # failure here indicates that the environment doesn't exist either
+            # because of incorrect naming syntax or because the env is still
+            # being built
+            self.page.get_by_role("combobox").nth(1).select_option(
+                f'{{"name":"{kernel}"}}'
+            )
+            # click Select to close popup (deal with the two formats of this dialog)
+            try:
+                self.page.get_by_role("button", name="Select", exact=True).click()
+            except Exception:
+                self.page.locator("div").filter(has_text="No KernelSelect").get_by_role(
+                    "button", name="Select"
+                ).click()
+
+    def set_environment(self, kernel):
+        """Set environment of a jupyter notebook.
+
+        IMPORTANT: The focus MUST be on the notebook on which you want to set
+        the environment.
+
+        Conda environments may still be being built shortly after deployment.
+
+        Parameters
+        ----------
+        kernel: str
+            Name of kernel to set.
+
+        Returns
+        -------
+        None
+        """
+
+        popup = self._check_for_kernel_popup()
+        # if there is not a kernel popup, make it appear
+        if not popup:
+            self.page.get_by_text("Kernel", exact=True).click()
+            self.page.get_by_role("menuitem", name="Change Kernel…").get_by_text(
+                "Change Kernel…"
+            ).click()
+
+        self._set_environment_via_popup(kernel)
+
+    def clone_repo(self, git_url, branch=None, wait_for_completion=5):
+        """Clone a git repo into the jupyterlab file structure.
+
+        The terminal is a blackbox for the browser. We can't access any of the
+        displayed text, therefore we have no way of knowing if the commands
+        are done executing. For this reason, there is an unavoidable sleep
+        here that prevents playwright from moving on to ensure that the focus
+        remains on the Terminal until we are done issuing our commands.
+
+        Parameters
+        ----------
+        git_url: str
+            url of git repo (must be public, referenced as http rather than ssh
+            format, e.g. https://github.com/nebari-dev/nebari.git)
+        branch: bool or str
+            (Optional) branch to checkout from the repo. Defaults to None to use
+            the default branch.
+
+        Returns
+        -------
+        None
+        """
+        import datetime as dt
```

Comment:
Why is this a lazy import? Any concerns putting it at the top with all other imports?

Reply:
This one is an oversight.